### ISSUE TITLE
hyperparamter tuning with sgd algorithm

### ISSUE THREAD

**Issue by PseudoProgrammer** [PaddlePaddle/Paddle#1119]
Hyperparameters tuning is so expensive to train model with of tens of million samples，So
how to tune hyperparameters efficienctly？ 

My settings：
```
24 Settings(
 25     algorithm='sgd',
 26     learning_rate=0.01,
 27     learning_method = 'adagrad',
 28     ada_epsilon = 1e-6,
 29     ada_rou = 0.95,
 30     batch_size = 789,
 31     learning_rate_decay_a=0,
 32     learning_rate_decay_b=0,
 33     num_batches_per_send_parameter=1,
 34     num_batches_per_get_parameter=1,
 35   ) 
```

@pengli09  Could you give us some suggestions?

**Comment by pengli09** [comment]
Hyper-parameter tuning is highly task dependent. The following is a blog post by Ilya Sutskever who is one the most experienced deep learning scientist. I recommend this post strongly and I believe you will benefit a lot from it.
http://yyue.blogspot.hk/2015/01/a-brief-overview-of-deep-learning.html

**Comment by PseudoProgrammer** [comment]
Thank you for your advice！Additionaly，how to tune the specific hyper-parameters of adagrad based on your experience, this blog have only advised some general hyper-parameters in deep learning.Such as ada_rou, batch_size,learning_rate_decay_a

**Comment by PseudoProgrammer** [comment]
I have 70 millions samples,5 million features for binary classification task.I hit a bottleneck with 89.6% accuracy.
